### ISSUE TITLE
Unison refuses to copy SymbolicLink/Junction files as "special files"

### ISSUE THREAD

**Issue by stdedos** [bcpierce00/unison#603]
Instead, it copies their contents.
I am not sure what the behavior is on Linux, as I haven't tested it.

I understand the complications of forcing either option (even while doing "smart things", like e.g. "copy contents when content is outside the roots"), so I'd like to be able to specify what I'd like to do.
Granularly (e.g. per-file operation) or globally (command line flag/option).

**Comment by gdt** [comment]
Symlinks work fine on POSIX; I just synchronized the result of "ln -s /dev/null link-to-dev-null" across NetBSD and macOS and I got a symlink in the other replicas.    This seems correct to me.

I think your enhancement request is basically a dup of #356, and the consensus there was that it's a difficult issue.  So it's probably best brought up on the hackers list, if anyone wants to dig into preparing a proposed code change.
